internal/utils.go

```diff
@@ -326,23 +329,31 @@ func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]interface{
 // When index is 0, it won't attach any number. When useManifestName is true,
 // it returns the policyConf.Manifests[manifestGroupIndex].Name otherwise,
 // It returns policyConf.Name + index
```

Comment:
The description for the function needs an update. I was trying to think about how it works, so I drafted one as a starter:
```golang
// getConfigurationPolicyName returns the `ConfigurationPolicy` name based on the provided PolicyConfig.
// If `consoldationManifests` is true or `manifests[].name` is not provided, it falls back to `policies[].name`.
// When a name is assigned, it increments the associated counter, using it as a suffix if it's greater
// than one to create unique names. The `manifestNameCounter` can be set to nil to force the function to
// return `policies[].name` regardless of whether `manifests[].name` is set.
```

Reply:
love it